api/src/routes/users.rs

```diff
@@ -33,15 +33,15 @@ pub struct UserInsert {
 )]
 #[post("/users")]
 pub async fn post_user(db: web::Data<Database>, body: web::Json<UserPost>) -> HttpResponse {
```

Comment:
Along with this, I recommend when we initialize the DB with the first user, we set `is_admin = true` for them.

```suggestion
pub async fn post_user(
    db: web::Data<Database>,
    body: web::Json<UserPost>,
    admin_user: Option<web::ReqData<User>>
) -> HttpResponse {
if not admin_user.is_admin ... // Not authorized
```

Reply:
let me edit the user data model to include that